XUtils.h

```diff
@@ -63,4 +63,11 @@ char* xStrdup(const char* str) ATTR_NONNULL;
 
 char* xStrndup(const char* str, size_t len) ATTR_NONNULL;
 
+static inline bool double_equal(double a, double b) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
+   return a == b;
+#pragma GCC diagnostic pop
```

Comment:
```suggestion
   return fabs(a-b) < 1E-6;
```

Reply:
I really want to compare to literal zero (cause the value is used as divisor), so the hack just to silence the compiler.

Reply:
But given the context where this is used, it doesn't really make sense to allow for other arbitrarily small intervals to pass through either. As formulated with this suggestion you could resolve times as low as 1µs, which already is far below the time we need for one cycle through our code. Filtering small differences also avoids "exploding numbers" issues down the way.